pkg/kds/mux/zone_sync.go

```diff
@@ -0,0 +1,61 @@
+package mux
+
+import (
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core"
+)
+
+type FilterV2 interface {
+	InterceptServerStream(stream grpc.ServerStream) error
+	InterceptClientStream(stream grpc.ClientStream) error
+}
+
+type OnGlobalToZoneSyncConnectFunc func(stream mesh_proto.KDSSyncService_GlobalToZoneSyncServer, errorCh chan error)
+
+func (f OnGlobalToZoneSyncConnectFunc) OnGlobalToZoneSyncConnect(stream mesh_proto.KDSSyncService_GlobalToZoneSyncServer, errorCh chan error) {
+	f(stream, errorCh)
+}
+
+var clientLog = core.Log.WithName("kds-delta-client")
+
+type KDSSyncServiceServer struct {
+	globalToZoneCb OnGlobalToZoneSyncConnectFunc
+	filters        []FilterV2
+	mesh_proto.UnimplementedKDSSyncServiceServer
+}
+
+func NewKDSSyncServiceServer(globalToZoneCb OnGlobalToZoneSyncConnectFunc, filters []FilterV2) *KDSSyncServiceServer {
+	return &KDSSyncServiceServer{
+		globalToZoneCb: globalToZoneCb,
+		filters:        filters,
+	}
+}
+
+var _ mesh_proto.KDSSyncServiceServer = &KDSSyncServiceServer{}
+
+func (g *KDSSyncServiceServer) GlobalToZoneSync(stream mesh_proto.KDSSyncService_GlobalToZoneSyncServer) error {
+	for _, filter := range g.filters {
+		if err := filter.InterceptServerStream(stream); err != nil {
+			clientLog.Error(err, "closing KDS stream following a callback error")
+			return err
```

Comment:
why log and return err? Wouldn't just return wrapped err work? same with other errors in this func